flow/view_holder.cc

```diff
@@ -110,16 +111,17 @@ void ViewHolder::UpdateScene(scenic::Session* session,
     entity_node_ = std::make_unique<scenic::EntityNode>(session);
     opacity_node_ = std::make_unique<scenic::OpacityNodeHACK>(session);
     view_holder_ = std::make_unique<scenic::ViewHolder>(
-        session, std::move(pending_view_holder_token_), "Flutter SceneHost");
+        session, std::move(pending_view_holder_token_), "Flutter ViewHolder");
     opacity_node_->AddChild(*entity_node_);
     opacity_node_->SetLabel("flutter::ViewHolder");
     entity_node_->Attach(*view_holder_);
-    if (ui_task_runner_ && pending_view_holder_token_.value) {
-      ui_task_runner_->PostTask(
-          [bind_callback = std::move(pending_bind_callback_),
-           view_holder_id = view_holder_->id()]() {
-            bind_callback(view_holder_id);
-          });
+
+    // Inform the rest of Flutter about the view being created.
+    // As long as we do this before calling `Present` on the session,
+    // View-related messages on the platform thread will never be processed
+    // before this internal message does.
+    if (on_view_created_ && pending_view_holder_token_.value) {
```

Comment:
I see pending_view_holder_token_ was moved out earlier (L 114) to the ViewHolder ctor. Wouldn't this always be an empty value here?

Reply:
Oops, this was the source of a bug I struggling with during testing :)

Good catch :+1: